### PR TITLE
SetVolumeIntent and SetVolumePercentIntent

### PR DESCRIPTION
**Pull request by DFranzen**
Additions as discussed in issue #35

### CODE REVIEW

File: squeezealexa/main.py [hunk 1]
```diff
@@ -148,6 +148,54 @@ def on_current(self, intent, session, pid=None):
...
+            return self.smart_response(text=heading,
+                                       speech=desc)
+        self.get_server().set_volume(vol * 10, pid)
+        desc = "Volume set to %d" % vol
+        return self.smart_response(text="Set Volume",
+                                   speech=desc)
+
+    @handler.handle(Custom.SET_VOL_PERCENT)
+    def on_set_vol_percent(self, intent, session, pid=None):
+        srv = self.get_server()
```
Comment (declension) [inline squeezealexa/main.py]: And this one I guess

File: squeezealexa/main.py [hunk 2]
```diff
@@ -148,6 +148,54 @@ def on_current(self, intent, session, pid=None):
...
+        srv.refresh_status()
+        try:
+            vol = float(intent['slots']['Volume']['value'])
+            print_d("Extracted playlist slot: %d" % vol)
```
[the remark this answers is not in the document]
Comment (DFranzen) [reply squeezealexa/main.py]: Yes sorry

File: squeezealexa/main.py [hunk 3]
```diff
@@ -148,6 +148,54 @@ def on_current(self, intent, session, pid=None):
             heading = None
         return self.smart_response(text=heading, speech=desc)
 
+    @handler.handle(Custom.SET_VOL)
+    def on_set_vol(self, intent, session, pid=None):
+        srv = self.get_server()
+        srv.refresh_status()
+        try:
+            vol = float(intent['slots']['Volume']['value'])
+            print_d("Extracted volume slot: %d" % vol)
```
[the remark this answers is not in the document]
Comment (DFranzen) [reply squeezealexa/main.py]: I think in the SetVolumePercent intent int is fine grained enough. In the normal intent we could allow .1f to get a bit finer grain.

File: squeezealexa/main.py [hunk 4]
```diff
@@ -148,6 +148,54 @@ def on_current(self, intent, session, pid=None):
             heading = None
         return self.smart_response(text=heading, speech=desc)
 
+    @handler.handle(Custom.SET_VOL)
+    def on_set_vol(self, intent, session, pid=None):
+        srv = self.get_server()
+        srv.refresh_status()
```
[the remark this answers is not in the document]
Comment (DFranzen) [reply squeezealexa/main.py]: nope not needed. I did not understand where the pid came from and tried to get it myself

File: squeezealexa/main.py [hunk 5]
```diff
@@ -148,6 +148,54 @@ def on_current(self, intent, session, pid=None):
...
+                                       speech=desc)
+        if (vol > 10) or (vol < 0):
+            print_d("Volume value out of range: %d" % vol)
+            desc = "Select a volume value between 0 and 10"
+            heading = "Volume value out of range: %d" % vol
+            return self.smart_response(text=heading,
+                                       speech=desc)
+        self.get_server().set_volume(vol * 10, pid)
+        desc = "Volume set to %d" % vol
+        return self.smart_response(text="Set Volume",
```
[the remark this answers is not in the document]
Comment (DFranzen) [reply squeezealexa/main.py]: Yes you are right. Alexa even only beeps, if her own volume is changed

**Comment by DFranzen** [comment]
Will make new psuh request with modifications

**Pull request closed without merging**